### PR TITLE
Add assertions for the UTF8 SRPs of J9ROMClass

### PR DESCRIPTION
**Pull request by hangshao0**
Signed-off-by: hangshao <hangshao@ca.ibm.com>

### CODE REVIEW

**Comment by hangshao0** [comment]
I didn't walk all the romMethods and romFields to assert the UTF8's there. Doing so could have more impact on the performance.

**Comment by pshipton** [comment]
jenkins test all zlinux jdk8

File: runtime/shared_common/CacheMap.cpp [hunk 1]
```diff
@@ -7400,3 +7406,50 @@ SH_CacheMap::updateLocalHintsData(J9VMThread* currentThread, J9SharedLocalStartu
...
+			innerClassNames++;
+		}
+	}
+
+#if defined(J9VM_OPT_VALHALLA_NESTMATES)
+	Trc_SHR_Assert_True((UDATA)J9ROMCLASS_NESTHOSTNAME(romClass) < romClassEnd);
+
+	if (romClass->nestMemberCount > 0) {
+		J9SRP *nestMemberNames = J9ROMCLASS_NESTMEMBERS(romClass);
+		for (i = 0; i < (U_32)nestMemberCount; i++) {
```
Comment (pshipton) [inline runtime/shared_common/CacheMap.cpp]: nestMemberCount looks like an undeclared variable.

Comment (hangshao0) [reply runtime/shared_common/CacheMap.cpp]: Fixed. It is inside #ifdef so compiler didn't capture this.

**Comment by pshipton** [comment]
jenkins test all zlinux jdk11

**Comment by DanHeidinga** [comment]
@hangshao0 Can you add some details describing why this change was made?  Also, were any startup performance tests run?

**Comment by hangshao0** [comment]
> Can you add some details describing why this change was made?

I have added more details in the description. 

> Also, were any startup performance tests run?

No. I guess the performance impact is minimal. There are 3 + interfaceCount + innerClassCount + nestMemberCount of pointer comparisons added when writing a romClass to the shared cache.

**Comment by DanHeidinga** [comment]
Thanks @hangshao0

**Pull request merged**